### ISSUE TITLE
Cannot use "this" in a typescript function declaration

### ISSUE THREAD

**Issue by a88zach** [gajus/eslint-plugin-jsdoc#919]
## Expected behavior
A user should be able to document a function in Typescript that uses the "this" keyword

## Actual behavior
Conflicting rules prevent this from passing warnings.  
* When `@param this` is declared the warning is:
![Screen Shot 2022-10-19 at 10 08 32 AM](https://user-images.githubusercontent.com/1724822/196714803-8caa3e33-0c97-4119-a592-67759529aa3e.png)

* When `@param this` is removed the warning is:
![Screen Shot 2022-10-19 at 10 09 13 AM](https://user-images.githubusercontent.com/1724822/196715008-94c03d0b-5371-4058-80fd-20fa2ca92c93.png)

## ESLint Config
```js
module.exports = {
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:jsdoc/recommended',
  ],
  overrides: [],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint', 'jsdoc'],
  root: true,
  rules: {
    'jsdoc/require-param-type': 0,
    'jsdoc/require-returns-type': 0,
    'jsdoc/no-types': 2,
  },
};
```

## ESLint sample

![Screen Shot 2022-10-19 at 10 08 32 AM](https://user-images.githubusercontent.com/1724822/196715453-0e09b13e-15e3-4a35-aa67-a9436e70344e.png)

```ts
type T = { name: string };

/**
 * @param this desc
 * @param bar number to return
 * @returns number returned back to caller
 */
function foo(this: T, bar: number): number {
  console.log(this.name);
  return bar;
}

const bound = foo.bind({ name: 'baz' });

bound(2);
```

![Screen Shot 2022-10-19 at 10 09 13 AM](https://user-images.githubusercontent.com/1724822/196715576-d3155410-7192-464e-a852-f19e44f70fdc.png)

```ts
type T = { name: string };

/**
 * @param bar number to return
 * @returns number returned back to caller
 */
function foo(this: T, bar: number): number {
  console.log(this.name);
  return bar;
}

const bound = foo.bind({ name: 'baz' });

bound(2);
```

## Environment

- Node version: 16
- ESLint version 8.25.0
- `eslint-plugin-jsdoc` version: 39.3.14

**Comment by brettz9** [comment]
I've resumed tolerating `@param this`, but we really still need to support `this: xyz` being ignored as a param in `require-param` if the user is documenting it instead with the (also [TS-sanctioned](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#this)) `@this`.

**Comment by brettz9** [comment]
I've gone ahead and also made `require-param` more tolerant in accepting `@param this`. However, since TS supports `@this`, it might be a better idea to ensure that the `this` object is documented with `@this` rather than as a param.
